### PR TITLE
test(melange): show dependency missing bug with virtual libraries

### PR DESCRIPTION
**Pull request by anmonteiro**
reproduces https://github.com/melange-re/melange/issues/1105

### CODE REVIEW

**Review by jchavarri — COMMENTED** [review]
ftr [this](https://mermaid.live/edit#pako:eNplkDtvhDAQhP8K2hoQjmUeLiIdd22qpAq-wjLmIPIDGaPkAvz3GCiz1Wjm0-xqFxC2lUChU_Zb9Nz5qP5gJgpTN8Jqbc09SpLX6NKMXN3P5Lrr5Gs6k3XQo5JaGj-t0eUkbo3Ugz_z-r-1it7aSQb-ygzEoKXTfGjDGcsOM_B9KGRAg2xlx2flGTCzBZTP3r4_jQDq3SxjmMeWe3kb-MNxDbTjagruyA3QBX6A4heU4orgkmCUk7IiRQxPoKgs0gyhKiurHGd5gfEWw6-1oSJLSQyyHbx1b-dvjhcdpZ8HsW_e_gAmLGBn) is the dep scheme that is reproduced (including the suggested simplification):

imagen

I think the test would be easier to understand if the libs where renamed to more commonly used names:
- `pal` -> `vlib`
- `pal_js` -> `impl`
- `common` -> `foo`

**Comment by anmonteiro** [comment]
Thank you for giving this a good look. I've made the changes you suggested.

File: test/blackbox-tests/test-cases/melange/virtual-lib-transitive.t [hunk 1]
```diff
@@ -0,0 +1,50 @@
+Test a case of virtual libraries where libraries depend on the virtual library
+transitively
+
+  $ mkdir -p foo vlib vlib_impl
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
```
Comment (jchavarri) [inline test/blackbox-tests/test-cases/melange/virtual-lib-transitive.t:6]: Not sure if using the latest version of dune lang is desirable somehow.

**Review by jchavarri — APPROVED** [review]
Thanks!

File: test/blackbox-tests/test-cases/melange/virtual-lib-transitive.t (hunk 1 above)
Comment (anmonteiro) [reply test/blackbox-tests/test-cases/melange/virtual-lib-transitive.t:6]: should be fine.

**Pull request merged**